### PR TITLE
f3d: init at 1.0.1

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/f3d/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+stdenv.mkDerivation rec {
+  name = "f3d";
+  version = "1.0.1";
+
+  src = fetchgit {
+    url = "https://gitlab.kitware.com/f3d/f3d.git";
+    rev = "a63df3d4b67880b54f60b9b946ab0eaaad108b06";
+    sha256 = "0a6r0jspkhl735f6zmnhby1g4dlmjqd5izgsp5yfdcdhqj4j63mg";
+  };
```
Comment (OPNA2608) [inline pkgs/applications/graphics/f3d/default.nix:11]:
```suggestion
  src = fetchFromGitLab {
    domain = "gitlab.kitware.com";
    owner = "f3d";
    repo = "f3d";
    rev = "v${version}";
    sha256 = "0a6r0jspkhl735f6zmnhby1g4dlmjqd5izgsp5yfdcdhqj4j63mg";
  };
```

File: pkgs/applications/graphics/f3d/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchgit, cmake, vtk_9, libX11, libGL, Cocoa, OpenGL }:
+
+stdenv.mkDerivation rec {
+  name = "f3d";
```
Comment (OPNA2608) [inline pkgs/applications/graphics/f3d/default.nix:4]:
```suggestion
  pname = "f3d";
```

File: pkgs/applications/graphics/f3d/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchgit, cmake, vtk_9, libX11, libGL, Cocoa, OpenGL }:
```
Comment (OPNA2608) [inline pkgs/applications/graphics/f3d/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitLab, cmake, vtk_9, libX11, libGL, Cocoa, OpenGL }:
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/f3d/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/applications/graphics/f3d/default.nix]: Changed, thanks.

File: pkgs/applications/graphics/f3d/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/applications/graphics/f3d/default.nix]: Good catch, done.

File: pkgs/applications/graphics/f3d/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/applications/graphics/f3d/default.nix]: Changed.

**Review by OPNA2608 — APPROVED** [review]

**Comment by OPNA2608** [comment]
LGTM if it runs for you.

Tested on non-NixOS Linux (Ubuntu 18.04). `SIGABRT`s on startup due to libGL, `LD_LIBRARY_PATH`ing the system's libGL libraries works (afaik this is a known problem, see https://github.com/NixOS/nixpkgs/issues/9415). Not tested with any 3D model.
image

**Comment by bcdarwin** [comment]
I have opened some 3d models and it works fine (except that it's compiled without the -r flag for ray tracing due to VTK not being compiled with necessary support).  Same issue on non-NixOS Linux, but works on OS X.

**Pull request merged**